### PR TITLE
refactor: refactor the style-layer-list for layers based sections in style-panel

### PR DESCRIPTION
**Pull request by JayaKrishnaNamburu**
## Description

This PR refactors the values in style-panel that handles complex values. Properties like `box-shadow`, `text-shadow`, `filter`, `backdrop-filter`, `transition` handles multiples values which needs to be displayed as list. And then allow users to edit them using a content panel. This PR removes all the individual layers of each property and created a common layer list item. 

## Steps for reproduction

1. Add values to all the sections,
2. Re-order the layers between the values of each property.
3. Deploy / Export the project. 

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [x] made a self-review
- [x] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [x] tested locally and on preview environment (preview dev login: 5de6)

### CODE REVIEW

**Comment by kof** [comment]
- [ ] can't add 2 text shadow layers, adding second deletes the first

**Comment by kof** [comment]
- [ ] backdrop filter is still in the advanced panel by default, needs to be removed, we only have there things by default which are not in the base panel

**Comment by JayaKrishnaNamburu** [comment]
> * [ ]  can't add 2 text shadow layers, adding second deletes the first

Will take a look, i actually managed to add multiple of them in local 🤔 
Screenshot 2024-05-10 at 17 07 02

**Comment by kof** [comment]
now text-shadow works now, must have hit something special

**Review by kof — APPROVED** [review]
Do you need @TrySound 's review?

File: apps/builder/app/builder/features/style-panel/style-layers-list.tsx [hunk 1]
```diff
@@ -76,47 +110,87 @@ export const LayersList = <
...
+                        onPressedChange={() => handleHideLayer(index)}
+                        icon={
+                          layer?.hidden ? (
+                            <EyeconClosedIcon />
+                          ) : (
+                            <EyeconOpenIcon />
+                          )
+                        }
+                      />
+                    ) : null}
```
Comment (kof) [inline apps/builder/app/builder/features/style-panel/style-layers-list.tsx:181]: for the sake of consistency
```suggestion
                    ) : undefined}
```

File: apps/builder/app/builder/features/style-panel/style-layers-list.tsx [hunk 2]
```diff
@@ -76,47 +110,87 @@ export const LayersList = <
...
+    return updateLayer(property, newLayers, value, index, createBatchUpdate);
   };
 
   return (
     <CssValueListArrowFocus dragItemId={dragItemId}>
       <Flex direction="column" ref={sortableRefCallback}>
-        {layers.value.map((layer, index) => {
+        {value.value.map((layer, index) => {
           if (layer.type !== "tuple" && layer.type !== "function") {
             return null;
```
Comment (kof) [inline apps/builder/app/builder/features/style-panel/style-layers-list.tsx:132]:
```suggestion
            return;
```

**Comment by JayaKrishnaNamburu** [comment]
> Do you need @TrySound 's review?

Not mandatory actually, just moved things around and deleted duplicate code 👍🏽

**Pull request merged**